kubescaler/scaler/aws/cluster.py

```diff
@@ -668,6 +668,25 @@ def scale(self, count):
             ],
         )
 
+        # Wait for stack update to be complete. 
+        # Note the reason not to use waiter for `stack_update_complete` is that, the waiter throws error if stacks is in UPDATE_ROLLBACK_COMPLETE/PROGRESS state. 
+        # So, we're checking if the stack is in PROGRESS status in any scenario (Create/Update/Rollback). if it doesn't then it can only be in FAILED or COMPLETE STATE
+        while True:
+            stack_update = self.cf.describe_stacks(
+                StackName=self.workers_name
+            )
+            current_status = stack_update["Stacks"][0]["StackStatus"]
+            if "PROGRESS" in current_status:
+                print(f"The stack-{self.workers_name} is {current_status}")
+                time.sleep(5)
+                continue
+            else:
+                print(f"The stack-{self.workers_name} is {current_status}")
+                break
+            
+        ##TODO Feature Request
```

Comment:
Can you tell me about times that it failed and why? It wasn't the case that it eventually gave you more information from the API above?

Reply:
When we update stack, sometimes it goes to "ROLLBACK_UPDATE_COMPLETE" something like that state. However, we're polling for "stack_update_completion" using waiter. if that happens, the waiter throws an error. I'll be able to replicate the exact behavior and i can show you..

Reply:
ahh, so the ROLLBACK means that it failed, and it's reverting it back (and deleting). There is a config value we set somewhere in our stack setup that says if it fails we want to delete I think, here: https://github.com/converged-computing/kubescaler/blob/299110a3072440d637bb1f072e6dd4a41e167b2d/kubescaler/scaler/aws/cluster.py#L65-L67. But there are different phases of rollback (e.g., see here) https://stackoverflow.com/questions/46264889/how-do-i-update-a-cloudformation-stack-with-state-update-rollback-complete so I think what we should do for now (while you are testing) is to print these stages for debugging, and try to understand the cases when it fails vs. "some other kind of rollback." And likely we can read docs online to verify this. Then we can act on explicit cases, if that makes sense.

Reply:
yeah exactly. I've tested my approach for checking updates. so, if we prevent scaling up going over max_node_count, i think we're fine for now. if ther's new cases arrives where stack update fails, we can address that.